src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```
@@ -3296,4 +3309,86 @@ private function prepare_taxonomy_limit_schema( array $query_params ) {
 
 		return $query_params;
 	}
+
+	/**
+	 * Retrieves post counts for the post type.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_count() {
+		$counts = wp_count_posts( $this->post_type );
+		$data   = array();
+
+		if ( ! empty( $counts ) ) {
```

If this is empty, I think this should return a 404 error ( wp_error here ).

---

Not sure about that for the following reasons:

-  The `wp_count_posts` function [always returns an stdClass object,](https://developer.wordpress.org/reference/functions/wp_count_posts/) even when no posts exist, so an empty count is still a valid count result. 
- A 404 status code only indicates that the resource is missing. The resource exists, it performs a count, but returns zero values.
- The endpoint should maintain consistent response structure regardless of count values.

Unless I'm missing something in [wp_count_posts](https://developer.wordpress.org/reference/functions/wp_count_posts/), 200 OK, therefore, is the most appropriate response code for this case as it's still a structured response.